### ISSUE TITLE
Unit tests must not use/expect credentials from the environment

### ISSUE THREAD

**Issue by tseaver** [googleapis/python-pubsub#366]
```bash
$ env | grep GOOGLE && echo YES || echo NO
NO
$ git log -1
commit 469ebaa3c449c881089dfc657da5902c1d031803 (HEAD -> master, origin/master, origin/HEAD)
Author: Peter Lamut <plamut@users.noreply.github.com>
Date:   Fri Apr 2 09:26:10 2021 +0200

    chore: regenerate GAPIC layer with latest changes (#345)
$ nox -e unit-3.8
nox > Running session unit-3.8
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/unit-3-8
nox > pip install asyncmock pytest-asyncio
nox > pip install mock pytest pytest-cov
nox > pip install -e .
nox > py.test --quiet --junitxml=unit_3.8_sponge_log.xml --cov=google/cloud --cov=tests/unit --cov-append --cov-config=.coveragerc --cov-report= --cov-fail-under=0 tests/unit
........................................................................ [  8%]
........................................................................ [ 17%]
........................................................................ [ 26%]
........................................................................ [ 34%]
........................................................................ [ 43%]
........................................................................ [ 52%]
........................................................................ [ 61%]
........................................................................ [ 69%]
........................................................................ [ 78%]
........................................................................ [ 87%]
........................................................................ [ 96%]
...............................                                          [100%]
=============================== warnings summary ===============================
...

tests/unit/pubsub_v1/publisher/test_publisher_client.py: 4 warnings
tests/unit/pubsub_v1/subscriber/test_subscriber_client.py: 9 warnings
  /home/tseaver/projects/agendaless/Google/src/python-pubsub/.nox/unit-3-8/lib/python3.8/site-packages/google/auth/_default.py:70: UserWarning: Your application has authenticated using end user credentials from Google Cloud SDK without a quota project. You might receive a "quota exceeded" or "API not enabled" error. We recommend you rerun `gcloud auth application-default login` and make sure a quota project is added. Or you can use service accounts instead. For more information about service accounts, see https://cloud.google.com/docs/authentication/
    warnings.warn(_CLOUD_SDK_CREDENTIALS_WARNING)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
- generated xml file: /home/tseaver/projects/agendaless/Google/src/python-pubsub/unit_3.8_sponge_log.xml -


823 passed, 15 warnings in 13.94s
```

Unit tests should always pass explicit dummy credentials (e.g., see [`test_init`](https://github.com/googleapis/python-pubsub/blob/469ebaa3c449c881089dfc657da5902c1d031803/tests/unit/pubsub_v1/publisher/test_publisher_client.py#L54-L56)).

**Comment by jimfulton** [comment]
@tseaver I don't understand your `nox` output.  Did the tests pass for you? Or am I missing something.

**Comment by plamut** [comment]
I believe he is referring to the same/similar thing we discovered yesterday?

When the GOOGLE_APPLICATION_CREDENTIALS env variable was not set, some of the unit tests fail, even though they shouldn't.

> google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create credentials and re-run the application.

I think this happens when a client instance is merely instantiated, thus we should instantiate clients in tests with dummy credentials to fix this.

FWIW, I was not able to reproduce a warning when unsetting the env variable, I got test failures, too. On the other hand, @tseaver probably had some other account active that the tests tried to grab.
